### PR TITLE
Connection: Unset jetpack conection active plugins from sync callable checksum when sync is not active 

### PR DESCRIPTION
**Pull request by darssen**
Fixes https://github.com/Automattic/vulcan/issues/469

## Proposed changes:

* Ensure we unset ``$jetpack_callables_sync_checksum['jetpack_connection_active_plugins'] if the `jetpack_connection_active_plugins` option gets changed in wpcom via the no sync fallback. This way we ensure that if/when sync gets re-enabled we send the callable no matter.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/469
## Testing instructions:

Check steps in https://github.com/Automattic/vulcan/issues/469 and check that we get now consistent values.

### CODE REVIEW

File: projects/packages/connection/src/class-plugin-storage.php [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 namespace Automattic\Jetpack\Connection;
 
+use Jetpack_options;
```
Comment (sergeymitr) [inline projects/packages/connection/src/class-plugin-storage.php]: Doesn't seem to matter, but still just in case
```suggestion
use Jetpack_Options;
```

**Review by sergeymitr — COMMENTED** [review]
Hi @darssen 👋 
The checksum value got removed, and the plugin list got synced as expected 👍 

It took a few minutes though for Sync to get it synced, and the site is empty.
Is it likely to increase on huge sites which are constantly syncing something, might it end up in the end of the queue?
Or for example a site that activates Jetpack for the first time, and has a lot to sync.

We could try adding additional check, and run `self::update_active_plugins_wpcom_no_sync_fallback()` if Sync is active, but `jetpack_connection_active_plugins` checksum is missing.
That way we'll know that the site underwent the "no-sync -> has-sync" plugin activation, and there's no shame in sending some critical data right away.

**Comment by darssen** [comment]
> It took a few minutes though for Sync to get it synced, and the site is empty.
> Is it likely to increase on huge sites which are constantly syncing something, might it end up in the end of the queue?
> Or for example a site that activates Jetpack for the first time, and has a lot to sync.

The problem with callable is that they normally are only sent when there is other information that needs to be synced too. So if when testing you just waited for it, it would take a little bit more time. Adding a post right away for example should speed up the process.
Regarding initial sync, Full Sync and Incremental go through different routes.

> We could try adding additional check, and run self::update_active_plugins_wpcom_no_sync_fallback() if Sync is active, but jetpack_connection_active_plugins checksum is missing.
That way we'll know that the site underwent the "no-sync -> has-sync" plugin activation, and there's no shame in sending some critical data right away.

Since this is a fallback and the initial issue is a bit of an edge case, unless we observe there is an actual problem with the delay I would keep things as they are.

**Review by sergeymitr — APPROVED** [review]
That all sounds great.
I think we're good to merge it then 👍

**Pull request merged**